server/src/main/java/org/apache/druid/server/compaction/CompactionStatus.java

```diff
@@ -376,15 +416,92 @@ private CompactionStatus evaluate()
       }
     }
 
+    /**
+     * Evaluates the fingerprints of all fingerprinted candidate segments against the expected fingerprint.
+     * <p>
+     * If all fingerprinted segments have the expected fingerprint, the check can quickly pass as COMPLETE. However,
+     * if any fingerprinted segment has a mismatched fingerprint, we need to investigate further by adding them to
+     * {@link #unknownStateToSegments} where their compaction states will be analyzed.
+     * </p>
+     */
+    private CompactionStatus allFingerprintedCandidatesHaveExpectedFingerprint()
```

Comment:
i think this is missing adding properly fingerprinted segments to compacted segments which is used for stats?